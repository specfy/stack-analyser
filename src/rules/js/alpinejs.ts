import { register } from '../../register.js';

register({
  tech: 'alpinejs',
  name: 'Alpine.js',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'alpinejs' }],
});
