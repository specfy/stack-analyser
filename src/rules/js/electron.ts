import { register } from '../../register.js';

register({
  tech: 'electron',
  name: 'Electron',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'electron' }],
});
