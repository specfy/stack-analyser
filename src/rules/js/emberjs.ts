import { register } from '../../register.js';

register({
  tech: 'emberjs',
  name: 'Ember',
  type: 'tool',
  dependencies: [
    { type: 'npm', name: 'ember-cli' },
    { type: 'npm', name: 'ember' },
    { type: 'npm', name: 'ember-data' },
    { type: 'npm', name: 'ember-source' },
  ],
});
