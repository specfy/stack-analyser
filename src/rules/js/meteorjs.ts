import { register } from '../../register.js';

register({
  tech: 'meteorjs',
  name: 'Meteor',
  type: 'tool',
  files: ['.meteor'],
  dependencies: [
    { type: 'npm', name: 'meteor' },
    { type: 'npm', name: 'meteor-node-stubs' },
  ],
});
