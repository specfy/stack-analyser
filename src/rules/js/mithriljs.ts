import { register } from '../../register.js';

register({
  tech: 'mithriljs',
  name: 'Mithril',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'mithril' }],
});
