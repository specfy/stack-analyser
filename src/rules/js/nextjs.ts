import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'tool',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
