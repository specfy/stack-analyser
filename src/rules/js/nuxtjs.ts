import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'tool',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
