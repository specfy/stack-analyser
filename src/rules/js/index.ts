import './alpinejs.js';
import './angular.js';
import './bootstrap.js';
import './emberjs.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './koa.js';
import './meteorjs.js';
import './mithriljs.js';
import './nextjs.js';
import './nuxtjs.js';
import './preactjs.js';
import './prettier.js';
import './prisma.js';
import './puppeteer.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './solidjs.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './sveltejs.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
