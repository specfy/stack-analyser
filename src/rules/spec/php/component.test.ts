import { describe, expect, it } from 'vitest';

import { analyser } from '../../../analyser/index.js';
import type { AllowedKeys } from '../../../index.js';
import { rawList } from '../../../loader.js';
import { flatten } from '../../../payload/helpers.js';
import { FakeProvider } from '../../../provider/fake.js';

import '../../../autoload.js';
import type { ComposerJson } from './component.js';

describe('php', () => {
  it('should match everything', async () => {
    const dependencies: Record<string, string> = {};
    for (const item of rawList) {
      if (item.type !== 'dependency' || item.ref.type !== 'php') {
        continue;
      }
      dependencies['example' in item.ref ? item.ref.example : item.ref.name] =
        '0.0.0';
    }
    const composerJson: ComposerJson = {
      name: 'test',
      'require-dev': dependencies,
    };

    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['composer.json'],
        },
        files: {
          '/composer.json': JSON.stringify(composerJson),
        },
      }),
    });

    const merged = flatten(res, { merge: true });
    expect(Array.from(merged.techs).sort()).toMatchSnapshot();
    expect(Array.from(merged.dependencies).sort()).toMatchSnapshot();
  });

  it('should match nothing', async () => {
    const res = await analyser({
      provider: new FakeProvider({
        paths: {
          '/': ['composer.json'],
        },
        files: {
          '/composer.json': JSON.stringify({ foo: 'bar' }),
        },
      }),
    });
    const match: AllowedKeys[] = ['php'];
    expect(
      Array.from(flatten(res, { merge: true }).techs).sort()
    ).toStrictEqual(match);
  });
});
