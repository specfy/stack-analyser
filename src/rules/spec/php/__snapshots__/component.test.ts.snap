// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`php > should match everything 1`] = `
[
  "airtable",
  "algolia",
  "alibabacloud",
  "amplitude",
  "apache_couchdb",
  "apache_kafka",
  "auth0",
  "aws",
  "aws.dynamodb",
  "aws.elasticache",
  "aws.lambda",
  "aws.rds",
  "aws.s3",
  "aws.sns",
  "aws.sqs",
  "bigcommerce",
  "bitbucket",
  "brevo",
  "clickhouse",
  "cloudflare",
  "cockroachdb",
  "contenful",
  "couchbase",
  "crowdin",
  "datadog",
  "datastax",
  "docusign",
  "dropbox",
  "elasticsearch",
  "firebase",
  "firebase.firestore",
  "gcp",
  "gcp.aiplatform",
  "gcp.appengine",
  "gcp.bigquery",
  "gcp.bigtable",
  "gcp.cloudbuild",
  "gcp.cloudrun",
  "gcp.dataflow",
  "gcp.dataproc",
  "gcp.datastore",
  "gcp.dialogflow",
  "gcp.functions",
  "gcp.gce",
  "gcp.gcs",
  "gcp.kms",
  "gcp.language",
  "gcp.memorystore",
  "gcp.pubsub",
  "gcp.secretmanager",
  "gcp.spanner",
  "gcp.speech",
  "gcp.sql",
  "gcp.tasks",
  "gcp.translate",
  "gcp.vision",
  "github",
  "gitlab",
  "googleanalytics",
  "hashicorp_vault",
  "heroku",
  "hubspot",
  "influxdb",
  "jira",
  "klaviyo",
  "kubernetes",
  "laravel",
  "launchdarkly",
  "lemonsqueezy",
  "magento",
  "mailchimp",
  "mailjet",
  "meilisearch",
  "mixpanel",
  "mongodb",
  "newrelic",
  "okta",
  "openai",
  "oraclecloud",
  "ovh",
  "pagerduty",
  "php",
  "phppest",
  "phpstan",
  "phpunit",
  "postgresql",
  "postmark",
  "rabbitmq",
  "redis",
  "resend",
  "sendgrid",
  "sentry",
  "shopify",
  "slack",
  "squareup",
  "stripe",
  "symfony",
  "tencentcloud",
  "twigphp",
  "twilio",
  "woocommerce",
  "wordpress",
  "yii2",
  "yousign",
  "zookeeper",
]
`;

exports[`php > should match everything 2`] = `
[
  [
    "php",
    "adilbaig/pagerduty",
    "0.0.0",
  ],
  [
    "php",
    "alek13/slack",
    "0.0.0",
  ],
  [
    "php",
    "algolia/algoliasearch-client-php",
    "0.0.0",
  ],
  [
    "php",
    "algolia/scout-extended",
    "0.0.0",
  ],
  [
    "php",
    "alibabacloud/client",
    "0.0.0",
  ],
  [
    "php",
    "alibabacloud/sdk",
    "0.0.0",
  ],
  [
    "php",
    "androk/yousign-api",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/dynamo-db",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/lambda",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/rds-data-service",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/s3",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/simple-s3",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/sns",
    "0.0.0",
  ],
  [
    "php",
    "async-aws/sqs",
    "0.0.0",
  ],
  [
    "php",
    "atyagi/elasticache-laravel",
    "0.0.0",
  ],
  [
    "php",
    "auth0/auth0-php",
    "0.0.0",
  ],
  [
    "php",
    "auth0/login",
    "0.0.0",
  ],
  [
    "php",
    "automattic/woocommerce",
    "0.0.0",
  ],
  [
    "php",
    "aws/aws-php-sns-message-validator",
    "0.0.0",
  ],
  [
    "php",
    "aws/aws-sdk-php",
    "0.0.0",
  ],
  [
    "php",
    "baopham/dynamodb",
    "0.0.0",
  ],
  [
    "php",
    "bigcommerce/api",
    "0.0.0",
  ],
  [
    "php",
    "bitbucket/client",
    "0.0.0",
  ],
  [
    "php",
    "cartalyst/stripe",
    "0.0.0",
  ],
  [
    "php",
    "cloudflare/sdk",
    "0.0.0",
  ],
  [
    "php",
    "colinmollenhour/cache-backend-redis",
    "0.0.0",
  ],
  [
    "php",
    "contentful/contentful",
    "0.0.0",
  ],
  [
    "php",
    "contentful/contentful-management",
    "0.0.0",
  ],
  [
    "php",
    "couchbase/couchbase",
    "0.0.0",
  ],
  [
    "php",
    "crowdin/crowdin-api-client",
    "0.0.0",
  ],
  [
    "php",
    "csharpru/vault-php",
    "0.0.0",
  ],
  [
    "php",
    "datadog/dd-trace",
    "0.0.0",
  ],
  [
    "php",
    "datadog/php-datadogstatsd",
    "0.0.0",
  ],
  [
    "php",
    "datastax/php-driver",
    "0.0.0",
  ],
  [
    "php",
    "doctrine/couchdb",
    "0.0.0",
  ],
  [
    "php",
    "doctrine/lexer",
    "0.0.0",
  ],
  [
    "php",
    "docusign/esign-client",
    "0.0.0",
  ],
  [
    "php",
    "drewm/mailchimp-api",
    "0.0.0",
  ],
  [
    "php",
    "ekino/newrelic-bundle",
    "0.0.0",
  ],
  [
    "php",
    "elasticsearch/elasticsearch",
    "0.0.0",
  ],
  [
    "php",
    "encentcloud/tencentcloud-sdk-php",
    "0.0.0",
  ],
  [
    "php",
    "fastly/magento2",
    "0.0.0",
  ],
  [
    "php",
    "getbrevo/brevo-php",
    "0.0.0",
  ],
  [
    "php",
    "google/analytics-admin",
    "0.0.0",
  ],
  [
    "php",
    "google/analytics-data",
    "0.0.0",
  ],
  [
    "php",
    "google/appengine-php-sdk",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-ai-platform",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-bigquery",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-bigtable",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-build",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-compute",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-dataflow",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-dataproc",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-datastore",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-dialogflow",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-firestore",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-functions",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-functions-framework",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-kms",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-language",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-pubsub",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-redis",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-run",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-secret-manager",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-spanner",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-speech",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-sql-admin",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-storage",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-tasks",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-translate",
    "0.0.0",
  ],
  [
    "php",
    "google/cloud-vision",
    "0.0.0",
  ],
  [
    "php",
    "heroku/heroku-buildpack-php",
    "0.0.0",
  ],
  [
    "php",
    "hubspot/api-client",
    "0.0.0",
  ],
  [
    "php",
    "hubspot/hubspot-php",
    "0.0.0",
  ],
  [
    "php",
    "influxdata/influxdb-client-php",
    "0.0.0",
  ],
  [
    "php",
    "intouch/newrelic",
    "0.0.0",
  ],
  [
    "php",
    "johnpbloch/wordpress",
    "0.0.0",
  ],
  [
    "php",
    "jolicode/slack-php-api",
    "0.0.0",
  ],
  [
    "php",
    "klaviyo/magento2-extension",
    "0.0.0",
  ],
  [
    "php",
    "knplabs/github-api",
    "0.0.0",
  ],
  [
    "php",
    "kubernetes/php-client",
    "0.0.0",
  ],
  [
    "php",
    "laravel/cashier",
    "0.0.0",
  ],
  [
    "php",
    "laravel/framework",
    "0.0.0",
  ],
  [
    "php",
    "laravel/slack-notification-channel",
    "0.0.0",
  ],
  [
    "php",
    "launchdarkly/launchdarkly-php",
    "0.0.0",
  ],
  [
    "php",
    "launchdarkly/server-sdk",
    "0.0.0",
  ],
  [
    "php",
    "lemonsqueezy/laravel",
    "0.0.0",
  ],
  [
    "php",
    "lesstif/php-jira-rest-client",
    "0.0.0",
  ],
  [
    "php",
    "longlang/phpkafka",
    "0.0.0",
  ],
  [
    "php",
    "m4tthumphrey/php-gitlab-api",
    "0.0.0",
  ],
  [
    "php",
    "magento/composer",
    "0.0.0",
  ],
  [
    "php",
    "mailchimp/makerting",
    "0.0.0",
  ],
  [
    "php",
    "mailchimp/transactional",
    "0.0.0",
  ],
  [
    "php",
    "mailjet/laravel-mailjet",
    "0.0.0",
  ],
  [
    "php",
    "mailjet/mailjet-apiv3-php",
    "0.0.0",
  ],
  [
    "php",
    "maknz/slack",
    "0.0.0",
  ],
  [
    "php",
    "martin-georgiev/postgresql-for-doctrine",
    "0.0.0",
  ],
  [
    "php",
    "meilisearch/meilisearch-php ",
    "0.0.0",
  ],
  [
    "php",
    "mixpanel/mixpanel-php",
    "0.0.0",
  ],
  [
    "php",
    "mongodb/mongodb",
    "0.0.0",
  ],
  [
    "php",
    "nbj/cockroachdb-laravel",
    "0.0.0",
  ],
  [
    "php",
    "nmred/kafka-php",
    "0.0.0",
  ],
  [
    "php",
    "omnipay/stripe",
    "0.0.0",
  ],
  [
    "php",
    "openai-php/client",
    "0.0.0",
  ],
  [
    "php",
    "openai-php/laravel",
    "0.0.0",
  ],
  [
    "php",
    "oracle/oci-php-sdk",
    "0.0.0",
  ],
  [
    "php",
    "ovh/ovh",
    "0.0.0",
  ],
  [
    "php",
    "pestphp/pest",
    "0.0.0",
  ],
  [
    "php",
    "php-amqplib/php-amqplib",
    "0.0.0",
  ],
  [
    "php",
    "phpclassic/php-shopify",
    "0.0.0",
  ],
  [
    "php",
    "phpstan/phpstan",
    "0.0.0",
  ],
  [
    "php",
    "phpunit/phpunit",
    "0.0.0",
  ],
  [
    "php",
    "predis/predis",
    "0.0.0",
  ],
  [
    "php",
    "resend/resend-php",
    "0.0.0",
  ],
  [
    "php",
    "roots/wordpress",
    "0.0.0",
  ],
  [
    "php",
    "ruflin/Elastica",
    "0.0.0",
  ],
  [
    "php",
    "sendgrid/sendgrid",
    "0.0.0",
  ],
  [
    "php",
    "sendinblue/api-v3-sdk",
    "0.0.0",
  ],
  [
    "php",
    "sentry/sdk",
    "0.0.0",
  ],
  [
    "php",
    "sentry/sentry",
    "0.0.0",
  ],
  [
    "php",
    "sentry/sentry-laravel",
    "0.0.0",
  ],
  [
    "php",
    "sentry/sentry-symfony",
    "0.0.0",
  ],
  [
    "php",
    "shopify/shopify-api",
    "0.0.0",
  ],
  [
    "php",
    "sleiman/airtable-php",
    "0.0.0",
  ],
  [
    "php",
    "smi2/phpclickhouse",
    "0.0.0",
  ],
  [
    "php",
    "smile/elasticsuite",
    "0.0.0",
  ],
  [
    "php",
    "snc/redis-bundle",
    "0.0.0",
  ],
  [
    "php",
    "socialiteproviders/okta",
    "0.0.0",
  ],
  [
    "php",
    "sparkinfluence/zookeeper",
    "0.0.0",
  ],
  [
    "php",
    "spatie/dropbox-api",
    "0.0.0",
  ],
  [
    "php",
    "spatie/laravel-analytics",
    "0.0.0",
  ],
  [
    "php",
    "square/square",
    "0.0.0",
  ],
  [
    "php",
    "stripe/stripe-php",
    "0.0.0",
  ],
  [
    "php",
    "symfony/symfony",
    "0.0.0",
  ],
  [
    "php",
    "theiconic/php-ga-measurement-protocol",
    "0.0.0",
  ],
  [
    "php",
    "twig/twig",
    "0.0.0",
  ],
  [
    "php",
    "twilio/sdk",
    "0.0.0",
  ],
  [
    "php",
    "wildbit/postmark-php",
    "0.0.0",
  ],
  [
    "php",
    "wp-cli/wp-cli",
    "0.0.0",
  ],
  [
    "php",
    "yiisoft/yii2",
    "0.0.0",
  ],
  [
    "php",
    "zumba/amplitude-php",
    "0.0.0",
  ],
]
`;
